narwhals/_pandas_like/series.py

```diff
@@ -665,6 +665,16 @@ def mode(self: Self) -> Self:
         result.name = native_series.name
         return self._from_native_series(result)
 
+    def log(self: Self, base: float) -> Self:
+        import numpy as np  # ignore-banned-import()
+
+        return self._from_native_series(np.log(self._native_series) / np.log(base))
+
+    def log10(self: Self) -> Self:
+        import numpy as np  # ignore-banned-import()
+
+        return self._from_native_series(np.log10(self._native_series))
```

Comment:
numpy maintains the underlying dtype (numpy, nullable numpy or pyarrow)

Reply:
hmm not sure if this is true when starting from Int64[pyarrow]
```python
In [29]: s
Out[29]:
0       1
1    <NA>
2       3
dtype: int64[pyarrow]

In [30]: np.log(s)
Out[30]:
0    0.000000
1         NaN
2    1.098612
dtype: float64
```

Reply:
Polars seems to **always** go for Float64. This should be doable. I am pushing a change for that.
Let me know if that's _too_ much hidden manipulation for the user.

Reply:
polars preserves float32:
```python
In [14]: pl.Series([1., 2.], dtype=pl.Float32).log10()
Out[14]:
shape: (2,)
Series: '' [f32]
[
        0.0
        0.30103
]
```

Reply:
polars preserves float32:
```python
In [14]: pl.Series([1., 2.], dtype=pl.Float32).log10()
Out[14]:
shape: (2,)
Series: '' [f32]
[
        0.0
        0.30103
]
```

Reply:
it looks like pandas preserves the backend for float - maybe we can do the cast to the appropriate type before calling numpy log?